packages/vue/src/components/molecules/SfDevice/SfDevice.vue

```diff
@@ -0,0 +1,109 @@
+<template>
+  <div class="sf-device">
+    <div class="sf-device__wrapper">
+      <div class="sf-device__morph" @click="clicked">
+        <div class="sf-device__screen">
+          <slot />
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+const devices = ["mobile", "tablet", "laptop", "detect"];
+
+export default {
+  name: "SfDevice",
+  props: {
+    device: {
+      type: String,
+      default: "detect",
+      validator: (propValue) => {
+        return devices.includes(propValue);
+      },
+    },
+    switchOnClick: {
+      type: Boolean,
+      default: false,
+    },
+    switchInterval: {
+      type: Number,
+      default: 0,
+    },
+  },
+  data: function () {
+    return {
+      i: 0,
+      morphEl: null,
+      intervalID: -1,
+      devices: devices,
+      classes: ["sf-device__mobile", "sf-device__tablet", "sf-device__laptop"],
+    };
+  },
+  mounted() {
+    this.morphEl = this.$el.querySelector(".sf-device__morph");
+
+    if (this.device === "detect") {
+      if (typeof window.matchMedia !== "undefined") {
+        const isMobile = window.matchMedia(
```

Comment:
suggestion: window property doesn't work with SSR - can you try to use CSS for conditional rendering? It's the best practice.

Reply:
If there were a way to trigger the media query manually without repeating the same CSS rules this might make sense but I couldn't think of a way to make it so the media query triggers one set of css rules but then can also be updated through JS to another set of css rules (for instance detecting laptop with pure css and then switching to mobile on click). I think keeping the check in JS is going to remain the best/only option.